### DISCUSSION TITLE
Any solution to separe gnd from D1mini

### DISCUSSION THREAD

**Discussion by gepetto666** [syssi/esphome-jk-bms#218] [General]
Hi.
i have powered the d1mini from thebateri with a DC DC converter. 24v to 5v.but there is a problem.
 ground from rs485 connector provide power to the esp even if i turnoff discharge from bms.
Is possible to use some optocupler to separate the ground?
Or other solution to separate data gnd from power gnd.
thnks
![bms](https://user-images.githubusercontent.com/49761965/203940639-011922f8-ecf1-4df8-9cb7-2703b319137f.jpg)

**Comment by syssi** [comment]
We depend on the community's experience here. I'm a software developer and cannot provide an informed/qualified answer here.

**Comment by paulsteigel** [comment]
> Hi. i have powered the d1mini from thebateri with a DC DC converter. 24v to 5v.but there is a problem. ground from rs485 connector provide power to the esp even if i turnoff discharge from bms. Is possible to use some optocupler to separate the ground? Or other solution to separate data gnd from power gnd. thnks !)

P- from JK-BMS is the Ground for the external load (when JK-BMS's discharge is set to on), when BMS is off, P- will be off too. Don't connect DC-DC converter's Ground to B- but P-. Doing this way, the DC-DC converter will only work if the BMS is on and Discharge is set to On!
I dont quite understand your point on separation of Ground!
